biloba/test/test_config.py

```diff
@@ -174,6 +174,50 @@ def test_set(self):
 
         self.assertEqual(conf['foo'], 'bar')
 
+    def test_set_dotted(self):
+        """
+        Set nested values using dot notation.
+        """
+        conf = config.Config()
+
+        conf['foo'] = {}
+        conf['foo.bar'] = 'baz'
+
+        self.assertEqual(conf['foo.bar'], 'baz')
+
+    def test_set_dotted_multiple(self):
+        """
+        Set nested values using dot notation (multiple dots).
+        """
+        conf = config.Config()
+
+        conf['foo'] = {'bar': {}}
+        conf['foo.bar.baz'] = 'pies'
+
+        self.assertEqual(conf['foo.bar.baz'], 'pies')
```

Comment:
I know for a fact it doesn't work, thought about trying to fix it.  The problem is that each `__getitem__` calls `expand`, which returns a new dict object so updating it doesn't persist.